rust-src/random_oracle/src/lib.rs

```diff
@@ -56,6 +57,12 @@ impl Buffer for RandomOracle {
     fn result(self) -> Self::Result { self.0.finalize() }
 }
 
+impl PartialEq for RandomOracle {
+    fn eq(&self, other: &Self) -> bool {
+        self.split().get_challenge() == other.split().get_challenge()
+    }
+}
```

Comment:
I am not a fan of this instance, but I'm not sure we can easily do anything better. Presumably there is no way to check internal state directly?

Additionally, please derive `Eq` a well since this relation is total.

Reply:
We could add #[cfg(test)] here or use test().get_challenge everywhere.

Reply:
Implemented Eq, Sha3_265 does not export anything useful for implementing it without using finalize()